app/app.go

```diff
@@ -138,14 +138,20 @@ func New(options *Options) App {
 // LivenessHandler is the http handler for the Kubernetes liveness probe
 func LivenessHandler(jobsDB jobsdb.JobsDB) http.HandlerFunc {
 	return func(w http.ResponseWriter, _ *http.Request) {
-		_, _ = w.Write([]byte(getHealthVal(jobsDB)))
+		healthy, responsePayload := getHealthVal(jobsDB)
+		if !healthy {
+			http.Error(w, "Cannot connect to db", http.StatusServiceUnavailable)
+			return
+		}
+		_, _ = w.Write([]byte(responsePayload))
```

Comment:
What are we doing to introduce this change gracefully?

Reply:
`    Liveness:   tcp-socket :8080 delay=120s timeout=1s period=10s #success=1 #failure=3`

There is nothing to be changed with respect to probes as we are not concerned about the health end point but this will help with SemaText alerts which is an external service pinging the data-plane. We can discuss at later point in time on how to combine the liveness and readiness probes with the health endpoint

Reply:
please elaborate, I fail to understand the purpose of this change

Reply:
With this change, we want to improve the health endpoint to return 200 only when the db is up. We are also planning to add this to readiness probes so that traffic won't be routed unless the db is up

Reply:
the ticket mentions other things as well, e.g. jobsdb is ready as well (tables are there in case of writer)
I need some lessons on readiness & liveliness probes

Reply:
You can read a bit about probes here: https://medium.com/@AADota/kubernetes-liveness-and-readiness-probes-difference-1b659c369e17
I have a set up a meeting tomorrow to discuss this